### PR TITLE
feat(gh-183): add logging functionality

### PR DESCRIPTION
**Pull request by Qazalbash**
## Summary

Logging functionality for better verbose and messages. `flowMC.logger.logger` is a modified form of [`loguru.logger`](https://loguru.readthedocs.io/en/stable/api/logger.html) to work with JAX, especially under `jax.jit` transform [^1]. [`loguru`](https://github.com/Delgan/loguru) has different logging modes, making logs easy to read.

[^1]: https://github.com/jax-ml/jax/discussions/25685

```python
from flowMC.logger import enable_logging, logger

enable_logging()

logger.critical("This is a critical message")
logger.debug("This is a debug message")
logger.error("This is an error message")
logger.exception("This is an exception message")
logger.info("This is an info message")
logger.success("This is a success message")
logger.trace("This is a trace message")
logger.warning("This is a warning message")
```

```txt
2025-04-05 19:34:47.169 | CRITICAL | This is a critical message
2025-04-05 19:34:47.169 | DEBUG    | This is a debug message
2025-04-05 19:34:47.170 | ERROR    | This is an error message
2025-04-05 19:34:47.170 | ERROR    | This is an exception message
NoneType: None
2025-04-05 19:34:47.171 | INFO     | This is an info message
2025-04-05 19:34:47.171 | SUCCESS  | This is a success message
2025-04-05 19:34:47.171 | TRACE    | This is a trace message
2025-04-05 19:34:47.172 | WARNING  | This is a warning message
```

## Related To

- #183

### CODE REVIEW

**Comment by kazewong** [comment]
`loguru` looks like an interesting package and thanks for bringing it to my attention!

While I am positive of integrating `loguru` into the verbose system, I am not exactly sure about whether we need the extra logger class you provide here. The main reason is I am quite allergic to have non-pure function in jax-transformer function in general. While the jax team is trying to keep external callback as innocuous as possible, external callbacks in transformed function still have a history of modifying the behavior of the function and return unexpected results. For this reason, I think we will keep logging outside any jax transform function.

This means the logging should go into strategy level but not resource level. I will have a look of this PR and start migrating some of these logging functionality into the main program, which I believe it should still be very beneficial

**Comment by Qazalbash** [comment]
That seems reasonable, let me know if I can help you with that!

**Comment by Qazalbash** [comment]
But what about the idea of checking gradients and other values at runtime? Is there any other way for it?
